src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
-
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Triangle {
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  getArea: () => number;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if ((a && b && c) <= 0) {
+      throw new Error('All sides must be higher than zero');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
```

Comment:
A variable here would make it even clearer what the check is like `failsTriangleInequality` or something like that, but the message probably covers it